jsonrpc/src/main.ts

```diff
@@ -604,6 +632,12 @@ function _createMessageConnection(messageReader: MessageReader, messageWriter: M
 			traceSendingResponse(message, method, startTime);
 			messageWriter.write(message);
 		}
+		function deleteRequestToken(id: string) {
```

Comment:
ya. I wanted to do what you are suggesting here, but cancellation was notification so, there was no clear point where I can do "when this is done, I can delete the file."

so, I did what I did. but, I guess I can do "then" here?
https://github.com/microsoft/vscode-languageserver-node/blob/master/jsonrpc/src/main.ts#L1096

but regardless, file can always left out (ex, request finished and client raised cancellation after that since the token is from a user and we can't control whether they hold it after request came back and raise cancellation).

but at the end, when client is stopped, client (who owns the folder) will clean up the folder removing any left out files.

by the way, typescript never removes any individual cancellation file created. they just keep create a new one and only delete the whole folder when vscode closes.

Reply:
Actually  https://github.com/microsoft/vscode-languageserver-node/blob/master/jsonrpc/src/main.ts#L1096 is a good point. Even better poinbt is when we create the Promise. Regarding the token still being canceled. I think we need to capture the disposable here  https://github.com/microsoft/vscode-languageserver-node/blob/master/jsonrpc/src/main.ts#L1096 and when the reponse has finished remove the listener by running the Disposable

Reply:
@dbaeumer can you give me a bit more detail on this disposable suggestion? it is more on me not familiar with typescript disposable thing on event. I looked and kind of don't understand what I am supposed to do.